FPIS/Views/AddDesignation.cs

```diff
@@ -0,0 +1,197 @@
+            {
+                labelDesignation.ForeColor = System.Drawing.Color.Red;
+                labelDesignation.Text = "Department name cannot have numbers.";
+                _isDataValid = false;
+                return;
+            }
+        }
+
+        private void ValidateDesignationName()
+        {
+            string designationName = materialTextBoxDesignationName.Text.Trim();
+
+            if (designationName.Length == 0)
+            {
+                labelDesignation.ForeColor = System.Drawing.Color.Red;
+                labelDesignation.Text = "Designation name is required!";
+                _isDataValid = false;
+                return;
+            }
+
+            AppDbContext dbContext = new();
+            var designation = dbContext.Designations.FirstOrDefault(designation => designation.DesignationName == designationName);
+            dbContext.Dispose();
+
+            if (designation != null)
+            {
+                _isDataValid = false;
+                labelDesignation.ForeColor = System.Drawing.Color.Red;
+                labelDesignation.Text = "Designation already Created.";
+                return;
+            }
+
+            Regex regex = new(nameRegexPattern);
+
+            if (!regex.IsMatch(designationName))
+            {
+                labelDesignation.ForeColor = System.Drawing.Color.Red;
+                labelDesignation.Text = "Designation name cannot have numbers.";
+                _isDataValid = false;
+                return;
+            }
+        }
+
+        private void ValidateInputs()
+        {
+            ValidateDepartmentName();
+            ValidateDesignationName();
+
+        }
+
+        private void LoadDeparments()
+        {
+            try
+            {
+                AppDbContext dbContext = new();
+                var departmentName = from Department in dbContext.Departments
+                                     select Department;
+                materialComboBoxDepartmentName.DataSource = departmentName.ToList();
+                materialComboBoxDepartmentName.DisplayMember = "DepartmentName";
+                // materialComboBoxDepartmentName.Text = "";
+                dbContext.Dispose();
+            }
+            catch (Exception ex)
+            {
+                Console.WriteLine($"Error Loading Departments: {ex}");
+                MaterialMessageBox.Show(ex.ToString());
+            }
+        }
+
+        private void AddDesignation_Load(object sender, EventArgs e)
+        {
+           LoadDeparments();
+        }
+
+       
+        private void CreateDesignation()
+        {
+            btnAddDesignation.Enabled = false;
+            AppDbContext dbContext = new();
+            var departmentName = dbContext.Departments.Where(dpt => dpt.DepartmentName == materialComboBoxDepartmentName.Text).Single();
+            var department = dbContext.Departments.Single(dept => dept.Id == departmentName.Id);
```

Comment:
The `departmentName` variable already stores the department we need to store for the Designation so we don't actually need the `department` field. So perhaps you can just change the for the `departmentName` afterwards

![image](https://user-images.githubusercontent.com/104218489/214075328-7b688952-69bc-4b66-a42c-14f8092960bf.png)